### PR TITLE
Added an example sql connection for managed identity

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: collections/_development_standards/platform-integration.md [hunk 1]
```diff
@@ -168,6 +168,62 @@ public static class AddDatabaseExtension
...
+
+```csharp
+public static class DatabaseExtensions
+{
+    private const string AzureResource = "https://database.windows.net/";
+
+    public static void AddDatabaseRegistration(this ConfigurationExpression config, string environment, string sqlConnectionString)
+    {
+        config.For<IDbConnection>().Use($"Build IDbConnection", c => {
+            var azureServiceTokenProvider = new AzureServiceTokenProvider();
```
Comment (EwanNoble) [inline collections/_development_standards/platform-integration.md:214]: `AzureServiceTokenProvider` is from `Microsoft.Azure.Services.AppAuthentication` which is now marked as legacy [and isn't recommended for use](https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication#:~:text=Microsoft.Azure.Services,Identity%20Migration%20Guidance.). If we don't want to fix that in the example here it should probably be at least called out as a caveat in the hope someone implements the new package in future.

Comment (ChrisJWoodcock) [reply collections/_development_standards/platform-integration.md:214]: Added a note to call out legacy status of AzureServiceTokenProvider

Comment (CloudPlatformer) [reply collections/_development_standards/platform-integration.md:214]: @ChrisJWoodcock Can the note include a link to https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication?

Comment (CloudPlatformer) [reply collections/_development_standards/platform-integration.md:214]: Added link to https://docs.microsoft.com/en-us/dotnet/api/overview/azure/app-auth-migration

**Review by notquitecraig — APPROVED** [review]

**Pull request merged**